upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -445,9 +447,26 @@ func (tf *TemplateFunctions) KopsControllerConfig() (string, error) {
 				if ig.Spec.Role == kops.InstanceGroupRoleNode {
 					profile, err := tf.LinkToIAMInstanceProfile(ig)
 					if err != nil {
-						return "", fmt.Errorf("getting role for ig %s: %v", ig.Name, err)
+						return "", fmt.Errorf("getting profile for ig %s: %v", ig.Name, err)
+					}
+					// The IAM Instance Profile has not been created at this point if it is not specified.
+					// Because the IAM Instance Profile and the IAM Role are created in IAMModelBuilder tasks.
+					// Therefore, the IAM Role associated with IAM Instance Profile is acquired only when it is not specified.
+					if ig.Spec.IAM != nil && ig.Spec.IAM.Profile != nil {
+						fmt.Printf("[h3poteto debug1] %+v\n", *profile.Name)
+						fmt.Printf("[h3poteto debug2] %+v\n", tf.cloud)
+						roles, err := tf.cloud.GetRolesInInstanceProfile(*profile.Name)
```

Comment:
Rather than adding `GetRolesInInstanceProfile` to the cloud interface and all of the cloud provider implementations, it's probably easier to add `GetRolesInInstanceProfile` to the AWSCloud interface and use type assertions on `tf.cloud` to get an AWSCloud, like this:

https://github.com/kubernetes/kops/blob/8c5dc5f0c96132ae14a8fd128ec4aa8b0386927f/pkg/resources/aws/vpc.go#L32

Reply:
OK, I see.

Reply:
I fixed it.